backoff/_wait_gen.py

```diff
@@ -51,14 +50,15 @@ def decay(
     """
     # Advance past initial .send() call
     yield  # type: ignore[misc]
+
     t = 0
-    while True:
+    a = initial_value
+    while a > min_value:
+        yield a
+        t += 1
         a = initial_value * math.e ** (-t * decay_factor)
-        if min_value is None or a > min_value:
-            yield a
-            t += 1
-        else:
-            yield min_value
+    while True:
+        yield min_value
```

Comment:
Following what's done in `expo`.

```suggestion
    a = initial_value
    while a > min_value:
        yield a
        a /= math.exp(decay_factor)
    while True:
        yield min_value
```